led_strip/examples/led_strip_spi_ws2812/main/led_strip_spi_ws2812_main.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Unlicense OR CC0-1.0
+ */
+#include <stdio.h>
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "led_strip.h"
+#include "esp_log.h"
+#include "esp_err.h"
+
+// GPIO assignment
+#define LED_STRIP_BLINK_GPIO  14
+
+// LED numbers in the strip
+#define LED_STRIP_LED_NUMBERS 24
+
+//#define CONFIG_BLINK_LED_RMT
+#define CONFIG_BLINK_LED_SPI
+
+static const char *TAG = "example";
+
+#define LED_STRIP_SPI_RES_HZ  (2.5 * 1000 * 1000)
+// 2.5MHz resolution, 1 tick = 0.4us (led strip needs a high resolution)
+
+// LED Strip object handle
+led_strip_handle_t led_strip;
```

Comment:
not a fan of using global variables. why not return the led_strip handle by the `configure_led` function?